test/framework/gardener_utils.go

```diff
@@ -0,0 +1,412 @@
+		newShoot := &gardencorev1beta1.Shoot{}
+		err = f.GardenClient.Client().Get(ctx, client.ObjectKey{Namespace: shoot.Namespace, Name: shoot.Name}, newShoot)
+		if err != nil {
+			f.Logger.Infof("Error while waiting for shoot to be reconciled: %s", err.Error())
+			return retry.MinorError(err)
+		}
+		shoot = newShoot
+		if ShootCreationCompleted(shoot) {
+			return retry.Ok()
+		}
+		f.Logger.Infof("Waiting for shoot %s to be reconciled", shoot.Name)
+		if newShoot.Status.LastOperation != nil {
+			f.Logger.Debugf("%d%%: Shoot State: %s, Description: %s", shoot.Status.LastOperation.Progress, shoot.Status.LastOperation.State, shoot.Status.LastOperation.Description)
+		}
+		return retry.MinorError(fmt.Errorf("shoot %q was not successfully reconciled", shoot.Name))
+	})
+}
+
+// AnnotateShoot adds shoot annotation(s)
+func (f *GardenerFramework) AnnotateShoot(shoot *gardencorev1beta1.Shoot, annotations map[string]string) error {
+	shootCopy := shoot.DeepCopy()
+
+	for annotationKey, annotationValue := range annotations {
+		metav1.SetMetaDataAnnotation(&shootCopy.ObjectMeta, annotationKey, annotationValue)
+	}
+
+	if _, err := f.MergePatchShoot(shoot, shootCopy); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RemoveShootAnnotation removes an annotation with key <annotationKey> from a shoot object
+func (f *GardenerFramework) RemoveShootAnnotation(shoot *gardencorev1beta1.Shoot, annotationKey string) error {
+	shootCopy := shoot.DeepCopy()
+	if len(shootCopy.Annotations) == 0 {
+		return nil
+	}
+	if _, ok := shootCopy.Annotations[annotationKey]; !ok {
+		return nil
+	}
+
+	// start the update process with Kubernetes
+	delete(shootCopy.Annotations, annotationKey)
+
+	if _, err := f.MergePatchShoot(shoot, shootCopy); err != nil {
+		return err
+	}
+	return nil
+}
+
+// MergePatchShoot performs a two way merge patch operation on a shoot object
+func (f *GardenerFramework) MergePatchShoot(oldShoot, newShoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+	patchBytes, err := kutil.CreateTwoWayMergePatch(oldShoot, newShoot)
+	if err != nil {
+		return nil, fmt.Errorf("failed to patch bytes")
+	}
+
+	patchedShoot, err := f.GardenClient.GardenCore().CoreV1beta1().Shoots(oldShoot.GetNamespace()).Patch(oldShoot.GetName(), types.StrategicMergePatchType, patchBytes)
+	if err == nil {
+		*oldShoot = *patchedShoot
+	}
+	return patchedShoot, err
+}
+
+// GetCloudProfile returns the cloudprofile from gardener with the give name
+func (f *GardenerFramework) GetCloudProfile(ctx context.Context, name string) (*gardencorev1beta1.CloudProfile, error) {
+	cloudProfile := &gardencorev1beta1.CloudProfile{}
+	if err := f.GardenClient.Client().Get(ctx, client.ObjectKey{Name: name}, cloudProfile); err != nil {
+		return nil, errors.Wrap(err, "could not get Seed's CloudProvider in Garden cluster")
+	}
+	return cloudProfile, nil
+}
+
+// DumpState greps all necessary logs and state of the cluster if the test failed
```

Comment:
is it possible to turn the state dump off during development? I get that it is very helpful to troubleshoot in case of problems, though it makes it harder to find the test output during dev.

Reply:
added a flag to the disable dump which can be also specified in the configuration file